docs/BRIDGE_FLOW.md

```diff
@@ -0,0 +1,304 @@
+                // If msg_data.len is > 161 bytes, we must call `process_message()` on the receiving contract, forwarding the newly minted coins with the call.
+                match message_data.len {
+                    160 => {
+                        transfer(amount, contract_id(), message_data.to);
+                    },
+                    161 => {
+                        transfer(amount, contract_id(), message_data.to);
+                    },
+                    _ => {
+                        if let Identity::ContractId(id) = message_data.to {
+                            let dest_contract = abi(MessageReceiver, id.into());
+                            dest_contract.process_message {
+                                coins: amount,
+                                asset_id: contract_id().value,
+                            }(msg_idx);
+                        };
+                    },
+                }
+
+                log(DepositEvent {
+                    to: message_data.to,
+                    from: message_data.from,
+                    amount: amount,
+                });
+            }
+        }
+    }
+}
+```
+
+### L2 (Fuel) Withdrawal
+
+The withdrawal message can be more convoluted, bear with us. The user will start the process by signaling a withdrawal transaction to the [L2 Bridge Contract](../packages/fungible-token/bridge-fungible-token/src/bridge_fungible_token.sw), which will burn the tokens and generate a message to be relayed from the L2 to the L1, with recipient to the [L1 Bridge Contract](../packages/portal-contracts/contracts/messaging/gateway/FuelERC20Gateway.sol). This message will be relayed later on by means of the block committer and the [Message Portal](../packages/portal-contracts/contracts/fuelchain/FuelMessagePortal.sol). 
+
+Once the transaction has been included in the Fuel blockchain, the user will need to await two events:
+- First, the Fuel blockchain will need to close an epoch. The epoch will be committed to the L1 blockchain in the [Chain State contract](../packages/portal-contracts/contracts/fuelchain/FuelChainState.sol).
+- Then, once the epoch has been committed, it is needed to wait for its finalization.
+
+After finality has been reached for the epoch that includes the block where the withdrawal transaction was signaled, the user can request the proofs of inclusion of the transaction and message in the block epoch that was comitted to the L1. The user will send a transaction call to the [Message Portal](../packages/portal-contracts/contracts/fuelchain/FuelMessagePortal.sol) 's `relayMessage` function that will check the attached proofs and the finalization status of the epoch that the message belongs to, then proceed to unpack the payload of said message, that should contain an execution instruction to call `finalizeWithdrawal` in  the [L1 Bridge Contract](../packages/portal-contracts/contracts/messaging/gateway/FuelERC20Gateway.sol), releasing the L1 locked tokens.
```

Comment:
We could also add additional docs on how these proofs are structured via 2 distinct merkle proofs. One to prove that the block header is included in the history of the epoch, and the other to prove their withdrawal message was included in that proven header.